src/app/components/Table/useSearchParamsPagination.ts

```diff
@@ -0,0 +1,18 @@
+import { To, useSearchParams } from 'react-router-dom'
+
+export function useSearchParamsPagination(paramName: string) {
+  const [searchParams] = useSearchParams()
+  const selectedPage = parseInt(searchParams.get(paramName) ?? '1', 10)
```

Comment:
not sure it's worth adding a guard against NaN

Reply:
ah - what do you think of
```ts
if (isNaN(selectedPage)) throw new Error('400 Bad Request')
```